nixos/modules/services/misc/gitea.nix

```diff
@@ -400,12 +413,31 @@ in
         message = ''
           When creating a database via NixOS, the db user and db name must be equal!
           If you already have an existing DB+user and this assertion is new, you can safely set
-          `services.gitea.createDatabase` to `false` because removal of `ensureUsers`
+          `services.gitea.database.createDatabase` to `false` because removal of `ensureUsers`
           and `ensureDatabases` doesn't have any effect.
         '';
       }
     ];
 
+    services.gitea.database = lib.mkMerge [
+      (lib.mkIf cfg.database.createDatabase {
+        host = lib.mkDefault "127.0.0.1";
```

Comment:
I am slightly undecided if we could drop the moDefault here but since it would be an annoying breaking change for some I think we should just leave it as is for now